### PR TITLE
New Package: VirtualResolutions.m2

### PR DESCRIPTION
**Pull request by mahrud**
This package, [implemented](https://github.com/Macaulay2/Workshop-2018-Madison/commits/Virtual) during the 2018 M2 workshop in Wisconsin and IMA Coding Sprint in April 2019, provides tools for constructing and studying virtual resolutions for products of projective spaces. In particular, it implements a number of the methods for constructing virtual resolutions for products of projective spaces as introduced by Berkesch, Erman, and Smith in [arXiv:1703.07631](http://arxiv.org/abs/1703.07631). This package also contains methods for constructing curves in P^1xP^2, as these are a natural source for interesting virtual resolutions.

Building on the `TateOnProducts` package, this package also contains a function allowing one to compute the minimal elements of the multigraded Castelnuovo-Mumford regularity of a saturated module, as in [arXiv:math/0305214](http://arxiv.org/abs/math/0305214).

A few notes:
- The auxiliary file `Colon.m2` is intended to be turned into an independent package, at which point VirtualResolutions will simply include it as an exported package. (cc: @mikestillman)
- Two lines in the documentation depend on #906 to be merged first. If you think that PR might not be merged in this release, simply removing `multigraded ` from lines 38 and 52 of `docs.m2` is perfectly fine.

cc: @mikeloper, @aalmousa, @julietteBruce.

### CODE REVIEW

**Comment by DanGrayson** [comment]
Maybe we can get #906 pulled.  I made a comment there.

File: M2/Macaulay2/packages/VirtualResolutions/doc.m2 [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+     {\em Multigraded Castelnuovo-Mumford Regularity} (@{HREF("http://arxiv.org/abs/math/0305214","arXiv:math/0305214")}@) for the definition of multigraded regularity.)
+     Building on the TateOnProducts package, this package contains a function allowing one
+     to compute the minimal elements of the multigraded Castelnuovo-Mumford regularity of a $B$-saturated module.
+
+     Continuing the example from above, we see that $(2,0)$ is an element of the multigraded
+     regularity of $S/J$. From this we can compute a virtual resolution of $S/J$.
+    Example
+     multigradedRegularity(X, J)
+     vres = virtualOfPair(J, {{3,1}})
+     multigraded betti vres
```
Comment (DanGrayson) [inline M2/Macaulay2/packages/VirtualResolutions/doc.m2]: `multigraded` is undefined, so this example fails, even though #906 has been merge.

Comment (mahrud) [reply M2/Macaulay2/packages/VirtualResolutions/doc.m2]: I can't reproduce the issue, but since there isn't much time I removed `multigraded` for now.

**Comment by DanGrayson** [comment]
Hmm, now it works, so I put "multigraded" back into the documentation for you in d870db91a3b3f0f89d0ed2a91db45d9b3d46d5ee

**Comment by mahrud** [comment]
Thank you!

**Pull request merged**